src/main/java/gregtech/common/command/benchmark/CommandBenchmarkLookup.java

```diff
@@ -0,0 +1,278 @@
+            try {
+                trials = Integer.parseInt(args[0]);
+                if (trials <= 0) throw new NumberFormatException();
+            } catch (NumberFormatException ignored) {
+                throw new WrongUsageException("gregtech.command.benchmark.lookup.usage");
+            }
+        }
+        Object2ObjectOpenHashMap<RecipeMap<?>, ObjectArrayList<TrialResults>> nsTrialTimes = new Object2ObjectOpenHashMap<>();
+        Object2ObjectOpenHashMap<RecipeMap<?>, List<Recipe>> recipeLists = new Object2ObjectOpenHashMap<>();
+        AtomicInteger failureCount = new AtomicInteger();
+        GTLog.logger.info("[Benchmarking] Starting recipe lookup benchmarking...");
+        for (int i = 0; i < trials; i++) {
+            for (RecipeMap<?> map : RecipeMap.getRecipeMaps()) {
+                if (recipeLists.computeIfAbsent(map, m -> new ObjectArrayList<>(m.getRecipeList())).isEmpty()) continue;
+                nsTrialTimes.computeIfAbsent(map, m -> new ObjectArrayList<>())
+                        .add(trial(recipeLists.get(map), (v, it, f) -> {
+                            Recipe r = map.findRecipe(v, it, f);
+                            return r == null ? Collections.emptyList() : ImmutableList.of(r);
+                        }, failureCount));
+            }
+            if ((i > 1) && ((i & (i - 1)) == 0)) {
+                GTLog.logger.info("[Benchmarking] {}th trial complete...", i);
+            }
+        }
+        GTLog.logger.info("[Benchmarking] Benchmarking complete. Outputting results:");
+        Object2ObjectOpenHashMap<RecipeMap<?>, BenchmarkResults> resultsCache = new Object2ObjectOpenHashMap<>();
+        for (var entry : nsTrialTimes.entrySet()) {
+            BenchmarkResults results = composeResults(entry.getValue());
+            resultsCache.put(entry.getKey(), results);
+            GTLog.logger.info("[Benchmarking] Recipe Map {}, measurements in nanoseconds:",
+                    entry.getKey().getLocalizedName());
+            GTLog.logger.info("[Benchmarking] - Characteristic numbers for No Match: {}", results.noMatchTest());
+            GTLog.logger.info("[Benchmarking] - Characteristic numbers for One Match Exact: {}",
+                    results.oneMatchExactTest());
+            GTLog.logger.info("[Benchmarking] - Characteristic numbers for Three Match Excess: {}",
+                    results.threeMatchExcessTest());
+        }
+        sender.sendMessage(new TextComponentTranslation("gregtech.command.benchmark.lookup.success")
+                .setStyle(new Style().setColor(TextFormatting.GREEN)));
+        sender.sendMessage(
+                new TextComponentTranslation("gregtech.command.benchmark.lookup.failures", failureCount.get())
+                        .setStyle(new Style().setColor(TextFormatting.RED)));
+        try (FileWriter writer = new FileWriter("benchmark-lookup-results.csv")) {
+            writer.append("Recipe Map,Trial Type,Minimum,Q1,Median,Q3,Maximum\n");
```

Comment:
Any chance we can make this fancy and have evenly spaced entries, with separators like `|` between the entries? If to hard, feel free to ignore.